pkg/utils/k8sutil.go

```diff
@@ -421,6 +370,58 @@ func EnsureK8sResources(ns, name string, funcObj *spec.Function, client kubernet
 		},
 	}
 
+	if len(funcObj.Spec.Template.Spec.Containers) == 0 {
+		funcObj.Spec.Template.Spec.Containers = append(funcObj.Spec.Template.Spec.Containers, v1.Container{
+			Name:  name,
```

Comment:
You set .Name and .Image unconditionally later on, so no need to do that here as well.